### PR TITLE
Fix issue #3 re .tsx tileset loading

### PR DESCRIPTION
**Pull request by pgainullin**

### CODE REVIEW

**Comment by wolfenrain** [comment]
PR for https://github.com/flame-engine/flame_tiled/issues/3

File: lib/tiled.dart [hunk 1]
```diff
@@ -122,37 +141,42 @@ class Tiled {
...
-                  (tile.flips.horizontally ? tileSize.width : 0),
-              tile.y.toDouble() * tileSize.height +
-                  (tile.flips.vertically ? tileSize.height : 0),
-            ),
-            rotation: flips.angle * math.pi / 2,
-            scale: tileSize.width / tile.width,
-          );
+
+          if (tile.image == null) {
+            throw('Tile ${tile.x}:${tile.y} gid ${tile.gid} image is null');
```
Comment (luanpotter) [inline lib/tiled.dart]:
```suggestion
            throw 'Tile ${tile.x}:${tile.y} gid ${tile.gid} image is null';
```

File: lib/tiled.dart [hunk 2]
```diff
@@ -122,37 +141,42 @@ class Tiled {
...
+          } else {
```
Comment (luanpotter) [inline lib/tiled.dart]: you don't need the else here since the throw will halt

File: lib/tiled.dart [hunk 3]
```diff
@@ -88,17 +91,33 @@ class Tiled {
...
+    String file = await Flame.bundle.loadString('assets/tiles/$filename');
+    final parser = TileMapParser();
+
+    final String tsxSourcePath = _parseXml(file)
+        .rootElement
+        .children
+        .whereType<XmlElement>()
+        .firstWhere((element) => element.name.local == 'tileset', orElse: () => null)
+        ?.getAttribute('source');
+    if(tsxSourcePath != null) {
```
Comment (luanpotter) [inline lib/tiled.dart]:
```suggestion
    if (tsxSourcePath != null) {
```

File: lib/tiled.dart [hunk 4]
```diff
@@ -88,17 +91,33 @@ class Tiled {
...
-      final parser = TileMapParser();
-      return parser.parse(contents);
-    });
+  XmlDocument _parseXml(String input) => XmlDocument.parse(input);
+
+  Future<TileMap> _loadMap() async {
+    String file = await Flame.bundle.loadString('assets/tiles/$filename');
+    final parser = TileMapParser();
+
+    final String tsxSourcePath = _parseXml(file)
```
Comment (luanpotter) [inline lib/tiled.dart]:
```suggestion
    final tsxSourcePath = _parseXml(file)
```

File: lib/tiled.dart [hunk 5]
```diff
@@ -88,17 +91,33 @@ class Tiled {
 
   Future _load() async {
     map = await _loadMap();
-    image = await Flame.images.load(map.tilesets[0].image.source);
+    if (map.tilesets[0].image != null)
+      image = await Flame.images.load(map.tilesets[0].image.source);
```
Comment (luanpotter) [inline lib/tiled.dart]:
```suggestion
    if (map.tilesets[0].image != null) {
      image = await Flame.images.load(map.tilesets[0].image.source);
    }
```

**Comment by spydon** [comment]
@pgainullin are you up for updating this PR? :)

**Comment by pgainullin** [comment]
> @pgainullin are you up for updating this PR? :)

sure, just updated incorporating @luanpotter's suggestions. However I have not tested it at all as I haven't had the time to sort through the dependency hell this triggers on my machine with Dart SDK v 2.13.0

**Review by erickzanardo — APPROVED** [review]

**Review by luanpotter — APPROVED** [review]
LGTM

**Review by spydon — APPROVED** [review]

**Pull request merged**